pkg/http/core/kubernetes/deploy.go

```diff
@@ -134,18 +133,19 @@ func Deploy(c *gin.Context, dm DeployManifestRequest) {
 			return
 		}
 
-		err = kc.VersionVolumes(u, dm.RequiredArtifacts)
+		namespace := u.GetNamespace()
+		err = kc.VersionVolumes(u, namespace, application, client)
+
 		if err != nil {
 			clouddriver.Error(c, http.StatusInternalServerError, err)
 			return
 		}
 
 		if kc.IsVersioned(u) {
-			kind := strings.ToLower(u.GetKind())
-			namespace := u.GetNamespace()
 			labelSelector := metav1.LabelSelector{
 				MatchLabels: map[string]string{
-					kubernetes.LabelKubernetesName: application,
+					kubernetes.LabelKubernetesName:      application,
+					kubernetes.LabelKubernetesManagedBy: kubernetes.Spinnaker,
 				},
 				MatchExpressions: []metav1.LabelSelectorRequirement{
```

Comment:
Can we move all the logic inside this if statement to a helper function that takes in whatever arguments it needs?
```go
if kc.IsVersioned(u) {
  err = version(u, ...) // Assuming that argument length doesn't exceed three.
  if err != nil {
    // Handle error.
  }
}